MapboxNavigation/CarPlayManager.swift

```diff
@@ -329,18 +329,23 @@ extension CarPlayManager: CPInterfaceControllerDelegate {
         }
     }
     public func templateWillDisappear(_ template: CPTemplate, animated: Bool) {
+        typealias TemplateBooleanClosure = (CPTemplate) -> Bool
+        guard let interface = interfaceController else { return }
+        
+        let listOrSearchOrMap = [CPSearchTemplate.self, CPListTemplate.self, CPMapTemplate.self]
+        let goingToSearch: TemplateBooleanClosure = { type(of: $0) == CPSearchTemplate.self }
```

Comment:
Nit: This closure is only used in one place; I think it would be fine to inline the equality check, as before.